airbyte-persistence/job-persistence/src/main/java/io/airbyte/persistence/job/DefaultJobPersistence.java

```diff
@@ -1146,4 +1154,53 @@ private static Table<Record> getTable(final String schema, final String tableNam
     return DSL.table(String.format("%s.%s", schema, tableName));
   }
 
+  public static void main(String[] args) {
+    final String url = "jdbc:postgresql://34.172.209.107:5432/postgres";
+    final String user = "postgres";
+    final String password = "";
+
+    Connection conn;
+    try {
+      conn = DriverManager.getConnection(url, user, password);
+      conn.setAutoCommit(false);
+
+      log.info("Connected to the PostgreSQL server successfully.");
+
+      var numRecs = 15_000_000;
+      PreparedStatement st = conn.prepareStatement("SELECT * FROM towns limit ?");
+      st.setInt(1, numRecs);
+      st.setFetchSize(1_500_000);
+
+      var start = System.currentTimeMillis();
+      // 1,000,000 is roughly equal to 660MB, so 1.5 mil is roughly 1GB.
+      log.info("Starting query...");
+      ResultSet rs = st.executeQuery();
+
+      var currRecs = 0;
+      log.info("Starting loop...");
+      while (rs.next()) {
```

Comment:
I wonder if having a cursor for another record actually means that it's trying to be parsed.  Maybe call .toString() on the record or something?